File: hotel/backend/src/main/kotlin/loshica/api/hotel/repositories/BookingRepository.kt
package loshica.api.hotel.repositories

import loshica.api.hotel.models.Booking
import loshica.api.hotel.models.User
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface BookingRepository : CrudRepository<Booking, Int> {
    fun findByCreatedBy(createdBy: User): Iterable<Booking>
}

File: hotel/frontend/src/repositories/BookingRepository.ts
import { Booking, BookingListModel, BookingPopulated } from '../abstractions/models';
import { APIRoute } from '../constants/enums';
import { Repository } from '../shared/decorators';
import CrudRepository from '../core/CrudRepository';

@Repository(APIRoute.BOOKINGS)
export default class BookingRepository extends CrudRepository<BookingPopulated, Booking> {
  async getByUser(userId: number): Promise<BookingListModel> {
    return (await this.api.get<BookingListModel>(`${this.route}?userId=${userId}`)).data;
  }
}
